### PR TITLE
docs(listbox): add kb for filtering

### PR DESCRIPTION
**Pull request by dimodi**

### CODE REVIEW

File: knowledge-base/listbox-filtering.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    * (optional) A Button to start the filtering. Alternatively, use the [`ValueChanged` or `OnChange` event of the TextBox]({%slug components/textbox/events%}).
+1. Create a [`DataSourceRequest` object](/blazor-ui/api/Telerik.DataSource.DataSourceRequest) and populate its `Filters` property with a single [`FilterDescriptor`](/blazor-ui/api/Telerik.DataSource.FilterDescriptor).
+    * If you need more complex filtering logic, use one or more [`CompositeFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor)s.
+1. Execute the [`ToDataSourceResult()` extension method]({%slug common-features-data-binding-onread%}#todatasourceresult-method) on the ListBox data. You will need to import the [`Telerik.DataSource.Extensions` namespace](/blazor-ui/api/Telerik.DataSource.Extensions).
+1. (optional) Show a [Loader]({%slug loader-overview%}) or a [LoaderContainer]({%slug loadercontainer-overview%}) during the filtering process.
+
+>tip If the filtering operator is fixed (for example, `Contains`), you can replace steps 2 and 3 with a standard LINQ expression:
+>
+> `ListBoxData = AllData.Where(x => x.Text.Contains(ListBoxFilterString)).ToList();`
```
Comment (dimodi) [inline knowledge-base/listbox-filtering.md:51]:
```suggestion
> `ListBoxData = AllData.Where(x => x.Name.Contains(ListBoxFilterString)).ToList();`
```

File: knowledge-base/listbox-filtering.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+## Solution
+
+1. Implement a user interface that includes:
+    * A [`TextBox`]({%slug components/textbox/overview%}) for the search string;
+    * (optional) A selection component (DropDownButton, DropDownList, ContextMenu, SplitButton) for the filter operator;
+    * (optional) A [Button]({%slug components/button/overview%}) to clear the TextBox `Value`.
+    * (optional) A Button to start the filtering. Alternatively, use the [`ValueChanged` or `OnChange` event of the TextBox]({%slug components/textbox/events%}).
+1. Create a [`DataSourceRequest` object](/blazor-ui/api/Telerik.DataSource.DataSourceRequest) and populate its `Filters` property with a single [`FilterDescriptor`](/blazor-ui/api/Telerik.DataSource.FilterDescriptor).
+    * If you need more complex filtering logic, use one or more [`CompositeFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor)s.
```
Comment (yordan-mitev) [inline knowledge-base/listbox-filtering.md:45]: You can use "one or more" with the singular form and avoid the strange "s" at the end.
```suggestion
    * If you need more complex filtering logic, use one or more [`CompositeFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor).
```

File: knowledge-base/listbox-filtering.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        public string Text { get; set; } = string.Empty;
+    }
+
+    #endregion Data Generation
+}
+````
+
+## See Also
+
+* [ToDataSourceResult Method]({%slug common-features-data-binding-onread%}#todatasourceresult-method)
```
Comment (yordan-mitev) [inline knowledge-base/listbox-filtering.md:231]: Use monospace for the methods as they are in a list
```suggestion
* [`ToDataSourceResult` Method]({%slug common-features-data-binding-onread%}#todatasourceresult-method)
```

File: knowledge-base/listbox-filtering.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+* [DataSourceRequest](/blazor-ui/api/Telerik.DataSource.DataSourceRequest)
```
Comment (yordan-mitev) [inline knowledge-base/listbox-filtering.md:232]:
```suggestion
* [`DataSourceRequest`](/blazor-ui/api/Telerik.DataSource.DataSourceRequest)
```

**Review by yordan-mitev — APPROVED** [review]

**Pull request merged**